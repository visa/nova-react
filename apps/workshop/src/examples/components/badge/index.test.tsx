/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { SubtleBadgeDefault } from './subtle-badge-default';
import { SubtleBadgeWithoutIcon } from './subtle-badge-without-icon';
import { SubtleBadgeWithIndicator } from './subtle-badge-with-indicator'; 
import { SubtleBadgeWithoutBackground } from './subtle-badge-without-background';

import { NeutralBadgeDefault } from './neutral-badge-default';
import { NeutralBadgeWithoutIcon } from './neutral-badge-without-icon';
import { NeutralBadgeWithIndicator } from './neutral-badge-with-indicator'; 
import { NeutralBadgeWithoutBackground } from './neutral-badge-without-background';
import { NeutralBadgeIcon } from './neutral-badge-icon';
import { NeutralBadgeWithIconWithoutBackground } from './neutral-badge-icon-without-background';

import { PositiveBadgeDefault } from './positive-badge-default';
import { PositiveBadgeWithoutIcon } from './positive-badge-without-icon';
import { PositiveBadgeWithIndicator } from './positive-badge-with-indicator'; 
import { PositiveBadgeWithoutBackground } from './positive-badge-without-background';
import { PositiveBadgeIcon } from './positive-badge-icon';
import { PositiveBadgeWithIconWithoutBackground } from './positive-badge-icon-without-background';

import { WarningBadgeDefault } from './warning-badge-default';
import { WarningBadgeWithoutIcon } from './warning-badge-without-icon';
import { WarningBadgeWithIndicator } from './warning-badge-with-indicator'; 
import { WarningBadgeWithoutBackground } from './warning-badge-without-background';
import { WarningBadgeIcon } from './warning-badge-icon';
import { WarningBadgeWithIconWithoutBackground } from './warning-badge-icon-without-background';

import { NegativeBadgeDefault } from './negative-badge-default';
import { NegativeBadgeWithoutIcon } from './negative-badge-without-icon';
import { NegativeBadgeWithIndicator } from './negative-badge-with-indicator'; 
import { NegativeBadgeWithoutBackground } from './negative-badge-without-background';
import { NegativeBadgeIcon } from './negative-badge-icon';
import { NegativeBadgeWithIconWithoutBackground } from './negative-badge-icon-without-background';

const examples = [

  { Component: SubtleBadgeDefault, title: metaData['subtle-badge-default'].title },
  { Component: SubtleBadgeWithoutIcon, title: metaData['subtle-badge-without-icon'].title },
  { Component: SubtleBadgeWithIndicator, title: metaData['subtle-badge-with-indicator'].title },
  { Component: SubtleBadgeWithoutBackground, title: metaData['subtle-badge-without-background'].title },

  { Component: NeutralBadgeDefault, title: metaData['neutral-badge-default'].title },
  { Component: NeutralBadgeWithoutIcon, title: metaData['neutral-badge-without-icon'].title },
  { Component: NeutralBadgeWithIndicator, title: metaData['neutral-badge-with-indicator'].title },
  { Component: NeutralBadgeIcon, title: metaData['neutral-badge-icon'].title },
  { Component: NeutralBadgeWithoutBackground, title: metaData['neutral-badge-without-background'].title },
  { Component: NeutralBadgeWithIconWithoutBackground, title: metaData['neutral-badge-icon-without-background'].title },

  { Component: PositiveBadgeDefault, title: metaData['positive-badge-default'].title },
  { Component: PositiveBadgeWithoutIcon, title: metaData['positive-badge-without-icon'].title },
  { Component: PositiveBadgeWithIndicator, title: metaData['positive-badge-with-indicator'].title },
  { Component: PositiveBadgeIcon, title: metaData['positive-badge-icon'].title },
  { Component: PositiveBadgeWithoutBackground, title: metaData['positive-badge-without-background'].title },
  { Component: PositiveBadgeWithIconWithoutBackground, title: metaData['positive-badge-icon-without-background'].title },

  { Component: WarningBadgeDefault, title: metaData['warning-badge-default'].title },
  { Component: WarningBadgeWithoutIcon, title: metaData['warning-badge-without-icon'].title },
  { Component: WarningBadgeWithIndicator, title: metaData['warning-badge-with-indicator'].title },
  { Component: WarningBadgeIcon, title: metaData['warning-badge-icon'].title },
  { Component: WarningBadgeWithoutBackground, title: metaData['warning-badge-without-background'].title },
  { Component: WarningBadgeWithIconWithoutBackground, title: metaData['warning-badge-icon-without-background'].title },

  { Component: NegativeBadgeDefault, title: metaData['negative-badge-default'].title },
  { Component: NegativeBadgeWithoutIcon, title: metaData['negative-badge-without-icon'].title },
  { Component: NegativeBadgeWithIndicator, title: metaData['negative-badge-with-indicator'].title },
  { Component: NegativeBadgeIcon, title: metaData['negative-badge-icon'].title },
  { Component: NegativeBadgeWithoutBackground, title: metaData['negative-badge-without-background'].title },
  { Component: NegativeBadgeWithIconWithoutBackground, title: metaData['negative-badge-icon-without-background'].title },
];

describe('Badge examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });
});
