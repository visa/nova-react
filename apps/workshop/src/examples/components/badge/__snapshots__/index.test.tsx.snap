// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge examples Default negative badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-critical"
  >
    <svg
      aria-hidden="true"
      aria-label="error"
      class="v-icon v-icon-visa v-icon-tiny v-icon-error"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M8 14C11.3137 14 14 11.3137 14 8C14 4.68629 11.3137 2 8 2C4.68629 2 2 4.68629 2 8C2 11.3137 4.68629 14 8 14ZM8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M9 4H7V8H9V4ZM8 10C7.44772 10 7 10.4477 7 11C7 11.5523 7.44772 12 8 12C8.55228 12 9 11.5523 9 11C9 10.4477 8.55228 10 8 10Z"
        fill-rule="evenodd"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
     Label
  </div>
</div>
`;

exports[`Badge examples Default neutral badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-neutral"
  >
    <svg
      aria-hidden="true"
      aria-label="information"
      class="v-icon v-icon-visa v-icon-tiny v-icon-information"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM7 5V3H9V5H7ZM6 9H7V10V11H6V13H7H9H10V11H9V10V8V7H6V9Z"
        fill-rule="evenodd"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
     Label
  </div>
</div>
`;

exports[`Badge examples Default positive badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-stable"
  >
    <svg
      aria-hidden="true"
      aria-label="success"
      class="v-icon v-icon-visa v-icon-tiny v-icon-success"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M14 8C14 11.3137 11.3137 14 8 14C4.68629 14 2 11.3137 2 8C2 4.68629 4.68629 2 8 2C8.74542 2 9.4591 2.13593 10.1176 2.38437L11.6319 0.870034C10.542 0.313733 9.30764 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16C12.4183 16 16 12.4183 16 8C16 7.21837 15.8879 6.46293 15.6789 5.74886L13.9678 7.37472C13.9891 7.58024 14 7.78885 14 8Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M15 3.46597L7.72838 10.9961L5 8.18244L6.42533 6.71257L7.72838 8.05635L13.5747 1.99609L15 3.46597Z"
        fill-rule="evenodd"
      />
    </svg>
     Label
  </div>
</div>
`;

exports[`Badge examples Default subtle badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-subtle"
  >
    <svg
      aria-hidden="true"
      aria-label="history"
      class="v-icon v-icon-visa v-icon-tiny v-icon-history"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M7 4H9V7.19575L10.47 6.277L11.53 7.973L7 10.8042V4Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M2 8C2 4.68629 4.68629 2 8 2C11.3137 2 14 4.68629 14 8C14 11.3137 11.3137 14 8 14V16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 10.3904 1.0484 12.5348 2.70861 14H1V16H6V11H4V12.4724C2.77174 11.3731 2 9.77615 2 8Z"
        fill-rule="evenodd"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
     Label
  </div>
</div>
`;

exports[`Badge examples Default warning badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-warning"
  >
    <svg
      aria-hidden="true"
      aria-label="warning"
      class="v-icon v-icon-visa v-icon-tiny v-icon-warning"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M8 0.492188L0 15.3493H16L8 0.492188ZM8 4.71071L3.34843 13.3493H12.6516L8 4.71071Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        d="M7 9.49219H9V11.4922H7V9.49219Z"
      />
    </svg>
     Label
  </div>
</div>
`;

exports[`Badge examples Negative badge with indicator should render correctly 1`] = `
<div>
  <div
    aria-label="error"
    class="v-badge v-badge-critical"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Negative badge without background should render correctly 1`] = `
<div>
  <div
    aria-label="error"
    class="v-badge v-badge-critical v-badge-clear"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Negative badge without icon should render correctly 1`] = `
<div>
  <div
    aria-label="error"
    class="v-badge v-badge-critical"
  >
    Label
  </div>
</div>
`;

exports[`Badge examples Negative icon badge with background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-critical"
  >
    <svg
      aria-hidden="true"
      aria-label="error"
      class="v-icon v-icon-visa v-icon-tiny v-icon-error-alt"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M9 1H7V11H9V1ZM8 15C8.55228 15 9 14.5523 9 14C9 13.4477 8.55228 13 8 13C7.44772 13 7 13.4477 7 14C7 14.5523 7.44772 15 8 15Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Badge examples Negative icon badge without background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-critical v-badge-clear"
  >
    <svg
      aria-hidden="true"
      aria-label="error"
      class="v-icon v-icon-visa v-icon-tiny v-icon-error-alt"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M9 1H7V11H9V1ZM8 15C8.55228 15 9 14.5523 9 14C9 13.4477 8.55228 13 8 13C7.44772 13 7 13.4477 7 14C7 14.5523 7.44772 15 8 15Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Badge examples Neutral badge icon without background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-neutral v-badge-clear"
  >
    <svg
      aria-hidden="true"
      aria-label="information"
      class="v-icon v-icon-visa v-icon-tiny v-icon-information"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM7 5V3H9V5H7ZM6 9H7V10V11H6V13H7H9H10V11H9V10V8V7H6V9Z"
        fill-rule="evenodd"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
  </div>
</div>
`;

exports[`Badge examples Neutral badge with indicator should render correctly 1`] = `
<div>
  <div
    aria-label="information"
    class="v-badge v-badge-neutral"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Neutral badge without background should render correctly 1`] = `
<div>
  <div
    aria-label="information"
    class="v-badge v-badge-neutral v-badge-clear"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Neutral badge without icon should render correctly 1`] = `
<div>
  <div
    aria-label="information"
    class="v-badge v-badge-neutral"
  >
    Label
  </div>
</div>
`;

exports[`Badge examples Neutral icon badge should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-neutral"
  >
    <svg
      aria-hidden="true"
      aria-label="information"
      class="v-icon v-icon-visa v-icon-tiny v-icon-information"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        clip-rule="evenodd"
        d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM7 5V3H9V5H7ZM6 9H7V10V11H6V13H7H9H10V11H9V10V8V7H6V9Z"
        fill-rule="evenodd"
      />
      <defs>
        <clippath>
          <rect
            height="16"
            width="16"
          />
        </clippath>
      </defs>
    </svg>
  </div>
</div>
`;

exports[`Badge examples Positive badge with indicator should render correctly 1`] = `
<div>
  <div
    aria-label="success"
    class="v-badge v-badge-stable"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Positive badge without background should render correctly 1`] = `
<div>
  <div
    aria-label="success"
    class="v-badge v-badge-stable v-badge-clear"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Positive badge without icon should render correctly 1`] = `
<div>
  <div
    aria-label="success"
    class="v-badge v-badge-stable"
  >
    Label
  </div>
</div>
`;

exports[`Badge examples Positive icon badge with background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-stable"
  >
    <svg
      aria-hidden="true"
      aria-label="success"
      class="v-icon v-icon-visa v-icon-tiny v-icon-checkmark"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        d="M15.7442 3.43912L5.03712 14.1462L0.330017 9.43912L1.74423 8.0249L5.03712 11.3178L14.33 2.0249L15.7442 3.43912Z"
      />
    </svg>
  </div>
</div>
`;

exports[`Badge examples Positive icon badge without background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-stable v-badge-clear"
  >
    <svg
      aria-hidden="true"
      aria-label="success"
      class="v-icon v-icon-visa v-icon-tiny v-icon-checkmark"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-primary"
        d="M15.7442 3.43912L5.03712 14.1462L0.330017 9.43912L1.74423 8.0249L5.03712 11.3178L14.33 2.0249L15.7442 3.43912Z"
      />
    </svg>
  </div>
</div>
`;

exports[`Badge examples Subtle badge with indicator should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-subtle"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Subtle badge without background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-subtle v-badge-clear"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Subtle badge without icon should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-subtle"
  >
    Label
  </div>
</div>
`;

exports[`Badge examples Warning badge with indicator should render correctly 1`] = `
<div>
  <div
    aria-label="warning"
    class="v-badge v-badge-warning"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Warning badge without background should render correctly 1`] = `
<div>
  <div
    aria-label="warning"
    class="v-badge v-badge-warning v-badge-clear"
  >
    <svg
      aria-hidden="true"
      class="v-icon v-icon-tiny v-ellipse"
      height="16"
      style="--v-icon-height: var(--size-scalable-8); --v-icon-width: var(--size-scalable-8);"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="8"
        r="8"
        style="fill: var(--v-badge-ellipse-color);"
      />
    </svg>
    Label
  </div>
</div>
`;

exports[`Badge examples Warning badge without icon should render correctly 1`] = `
<div>
  <div
    aria-label="warning"
    class="v-badge v-badge-warning"
  >
    Label
  </div>
</div>
`;

exports[`Badge examples Warning icon badge with background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-warning"
  >
    <svg
      aria-hidden="true"
      aria-label="warning"
      class="v-icon v-icon-visa v-icon-tiny v-icon-warning"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M8 0.492188L0 15.3493H16L8 0.492188ZM8 4.71071L3.34843 13.3493H12.6516L8 4.71071Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        d="M7 9.49219H9V11.4922H7V9.49219Z"
      />
    </svg>
  </div>
</div>
`;

exports[`Badge examples Warning icon badge without background should render correctly 1`] = `
<div>
  <div
    class="v-badge v-badge-icon v-badge-warning v-badge-clear"
  >
    <svg
      aria-hidden="true"
      aria-label="warning"
      class="v-icon v-icon-visa v-icon-tiny v-icon-warning"
      focusable="false"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="v-icon-secondary"
        clip-rule="evenodd"
        d="M8 0.492188L0 15.3493H16L8 0.492188ZM8 4.71071L3.34843 13.3493H12.6516L8 4.71071Z"
        fill-rule="evenodd"
      />
      <path
        class="v-icon-primary"
        d="M7 9.49219H9V11.4922H7V9.49219Z"
      />
    </svg>
  </div>
</div>
`;
